python/dask_cudf/dask_cudf/core.py

```diff
@@ -743,6 +751,29 @@ def from_dask_dataframe(df):
     return df.map_partitions(cudf.from_pandas)
 
 
+def _shuffle_context(func, *args, shuffle="tasks", **kwargs):
+    # Utility to check `shuffle` kwarg and use dask-config
+    # context for "explicit-comms" option, when necessary
+    if shuffle not in ("tasks", "explicit-comms"):
+        raise ValueError(
+            f"Dask-cudf only supports in-memory shuffling with "
+            f"'tasks' or 'explicit-comms'. Got shuffle={shuffle}"
+        )
+    if shuffle == "explicit-comms":
+        try:
+            import dask_cuda  # noqa: F401
+        except ImportError:
+            raise ValueError(
+                "shuffle='explicit-comms' requires dask_cuda. "
+                "Please install dask_cuda, or use shuffle='tasks'."
+            )
+        # Use explicit-comms shuffle
+        with dask.config.set({"explicit-comms": True}):
+            return func(*args, shuffle="tasks", **kwargs)
```

Comment:
For now, this PR is just leveraging the existing `dask.config` approach to activate explicit-comms. Does this seem reasonable @madsbk ?

Reply:
Yes

Reply:
I wonder if we should change the monkey-patching in `dask_cuda` to patch `dask.dataframe.shuffle.rearrange_by_column` to handle `shuffle="explicit-comms"`, rather than monkey-patching `rearrange_by_column_tasks`. Then as soon as `dask_cuda` is imported one can always just say `shuffle="explicit-comms"` and we don't need to wrap up everything individually.